### DISCUSSION TITLE
JSON in MSSQL Server

### DISCUSSION THREAD

**Discussion by mikkelwf** [prisma/prisma#18256] [Q&A] [answered]
According to the documentation, rudimental support for JSON in sqlserver is supported

https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#json

However trying to set `Json` with a `@db.NVarChar` fails the schema, it says `Error validating field `xxx` in model `XXX`: Field `xxx` in model `XXX` can't be of type Json. The current connector does not support the Json type.`

What to do?

**Comment by janpio** [comment, marked as answer]
Share the schema so we can reproduce and provide some more inforamtion what version of Prisma you are using please.

**Reply by mikkelwf** [reply]
```prisma
prisma                  : 4.11.0
@prisma/client          : 4.11.0
Current platform        : darwin-arm64

generator client {
  provider = "prisma-client-js"
  previewFeatures = ["clientExtensions"]
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model MetaRecipes {
  id                     Int      @id @default(autoincrement())
  recipe_id              Int?
  createdAt              DateTime
  updatedAt              DateTime
  lead_time              Json?  @db.NVarChar(Max)
}
```

**Reply by mikkelwf** [reply]
![Skærmbillede 2023-03-13 kl  10 23 43](https://user-images.githubusercontent.com/1061520/224659777-ecde82d1-2a5d-4781-8afe-4bb2aacd682d.png)

**Reply by nurul3101** [reply]
@mikkelwf SQL Server does not have a native JSON data type, but it does support storing and querying JSON data as text. 
You would need to change the type form `Json?` to `String?` and then it should work as expected

```prisma
model MetaRecipes {
  id        Int      @id @default(autoincrement())
  recipe_id Int?
  createdAt DateTime
  updatedAt DateTime
  lead_time String?  @db.NVarChar(1000)
}
```

You can then use these [functions](https://learn.microsoft.com/en-us/sql/relational-databases/json/json-data-sql-server?view=sql-server-ver15#extract-values-from-json-text-and-use-them-in-queries) to test/extract values from the JSON text column.

**Reply by mikkelwf** [reply]
I know I can work around it, and I already have.. 

And I know JSON is not natively supported in MSSQL.

But the Prisma documentation states that the Prisma schema type `Json` can be used with MSSQL, and maps natively with `@db.NVarChar`. 

This is obviously not the case.

I think the documentation is a bit misleading - it should clearly state that the Prisma type `Json` is not supported for MSSQL (the rest with working around the limitation is fine).

**Reply by janpio** [reply]
You are correct, our docs there were misleading. This PR will fix that: https://github.com/prisma/docs/pull/4584

Here is the issue for the feature request to implement support for `Json`: https://github.com/prisma/prisma/issues/7417

**Reply by james246** [reply]
I ran into this confusion today when looking at the docs, which imply JSON is supported in SQL Server. I found my way via Google to here which answered my question that it is not supported. 

For the benefit of others, having this PR merged would be good @janpio ! 👍
